### DISCUSSION TITLE
[QUESTION] Windows App (but don't need WinUI), SkiaSharp, OpenGL: Good starting point?

### DISCUSSION THREAD

**Discussion by ToolmakerSteve** [mono/SkiaSharp#2869] [Ideas]
Situation:
* Updating a cross-platform app (from .Net Framework and Xamarin Native to .Net 8) that uses SkiaSharp and OpenGL.
* Don't need native UI Widgets on any platform: it is all done via Skia or GL.

Maybe relevant: we currently use OpenTK 4.8.2. Mainly for its Math classes. On iOS, we added some sources from Xamarin.iOS's obsolete version of OpenTK, to use iOS's (deprecated) GLES API.  
Considering Silk.Net or Angle for the future. Especially if iOS ever drops the deprecated GLES API, forcing apps to target Metal.

On Windows, we don't need WinUI, Maui, WPF, WinForms - the various Windows UI platforms for which SkiaSharp offers a GL Control.  

---------------------------------------------
Question:
* On Windows, which of those UI targets would use the least system resources, if all we need is a GLSurface to which we can render Skia and GL? Or does it not matter much, so do whichever is convenient?   [For running on a Windows tablet.]

OR

* Would it make more sense in our case to create a Window with SkiaSharp + GL, that isn't any of those?  Sample code?  

----------------------------------------------
**UPDATE**

I see class SkiaSharp.Views.Windows.AngleSwapChainPanel.

Does that use Google Angle to support GLES on Direct3D?

Feels like a good match to where cross-platform 3D graphics are headed.

Any plan to use Angle (to Metal) on iOS, so SKGLView no longer depends on iOS GLKit.GLKView?

**Comment by nockawa** [comment]
Hey there, I have a similar situation, and so far I'm happy using WinForms (yes, you heard me, but the .net 8 version, of course), with SkiaSharp + GL `SKGLControl`.
Thing is:
 - WPF doesn't have GPU accelerated solution
 - If you want an ANGLE based, it looks possible by judging the source code of Avalonia, but it's hell of a work and all I saw everywhere for Windows target is OpenGL being used.

Vulkan and ANGLE are apparently used for other OSes...

Turns out that WinForms for .net 8 is pretty lightweight, so it suits my needs.
I would preferably have like a WPF + D3D version for the sake of hardware layer uniformity, but looks like quiet a job to complete because I couldn't find any existing implementation out there. And honestly, for this kind of usage (I come from the 3D video game industry), I think Open GL is more than ok.
